modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/IgniteH2Indexing.java

```diff
@@ -1090,14 +1091,14 @@ private List<? extends FieldsQueryCursor<List<?>>> executeDml(
             if (roEx != null) {
                 throw new IgniteSQLException(
                     "Failed to execute DML statement. Cluster in read-only mode [stmt=" + qryDesc.sql() +
-                    ", params=" + Arrays.deepToString(qryParams.arguments()) + "]",
+                    ", params=" + S.toString(QueryParameters.class, qryParams) + "]",
```

Comment:
If one of the query parameter is huge (say large CLOB/BLOB) usage of _Arrays.deepToString_ does two "bad" things which increases heap memory usage:

- creates full string representation of the large argument
- once an exception is logged this large string is passed to logging framework 

This can cause unneeded and unexpected OOM error just because of logging. The actual error is not logged in such case. Say I get the following error stack during testing:

```
[2024-08-29T22:03:37,692][ERROR][client-connector-#149%jdbc_thin_blob_test.JdbcThinLobTest.test_jdbc_thin_lob.blob_size.1073741824.clob_size.1024.server_heap.12.client_heap.8.ignite_version.dev%] ....
java.lang.OutOfMemoryError: null
at java.base/java.lang.AbstractStringBuilder.hugeCapacity(AbstractStringBuilder.java:214) ~[?:?]
at java.base/java.lang.AbstractStringBuilder.newCapacity(AbstractStringBuilder.java:206) ~[?:?]
at java.base/java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:173) ~[?:?]
at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:775) ~[?:?]
at java.base/java.lang.StringBuilder.append(StringBuilder.java:252) ~[?:?]
at java.base/java.util.Arrays.toString(Arrays.java:4988) ~[?:?]
at java.base/java.util.Arrays.deepToString(Arrays.java:5189) ~[?:?]
at java.base/java.util.Arrays.deepToString(Arrays.java:5161) ~[?:?]
at org.apache.ignite.internal.processors.query.h2.IgniteH2Indexing.executeDml(IgniteH2Indexing.java:1100) 
```

***

On the other hand the  _S.toString_ handles huge strings and huge byte arrays with care. It skips most of the data and replaces it with elipsis.   Like  [1, 2, ... and 1073741724 more]  for byte arrays and "aaa ... and 1041353 skipped ... aaa" for strings.

At the similar context SQL arguments are already logged this carefull way.  See for example the JdbcQueryExecuteRequest::toString.   If such a request is added into the log string the huge arguments are stripped.